internal/objectstore/fileobjectstore.go

```diff
@@ -260,6 +263,20 @@ func (t *fileObjectStore) loop() error {
 					err: t.remove(ctx, req.path),
 				}:
 				}
+
+			default:
+				return fmt.Errorf("unknown request type %d", req.op)
+			}
+
+		case <-timer.Chan():
+
+			// Reset the timer, as we've jittered the interval at the start of
```

Comment:
But you still have that problem even if you're jitted. We don't sequence other object stores (file-backed non-HA), so there is always a chance we can collide on a file system, but not on the same file. They're partitioned by directories (`objectstore/modeluuid`).

As for s3, we take the lock on prune deletion to prevent another object store within another controller for the same model-uiid from attempting to work on a given file.